drivers/spi/spi_ll_stm32.c

```diff
@@ -459,42 +459,73 @@ static void spi_stm32_cs_control(const struct device *dev, bool on)
 #endif
 }
 
-static void spi_stm32_complete(const struct device *dev, int status)
-{
-	const struct spi_stm32_config *cfg = dev->config;
-	SPI_TypeDef *spi = cfg->spi;
 #ifdef CONFIG_SPI_STM32_INTERRUPT
-	struct spi_stm32_data *data = dev->data;
-
+static void spi_stm32_disable_irqs(SPI_TypeDef *spi)
+{
 	ll_func_disable_int_tx_empty(spi);
 	ll_func_disable_int_rx_not_empty(spi);
 	ll_func_disable_int_errors(spi);
-#endif
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	LL_SPI_DisableIT_DXP(spi);
+	LL_SPI_DisableIT_TXTF(spi);
+	LL_SPI_DisableIT_EOT(spi);
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi) */
+}
+#endif /* CONFIG_SPI_STM32_INTERRUPT */
 
+static void spi_stm32_clear_status(SPI_TypeDef *spi)
+{
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_spi_fifo)
 	/* Flush RX buffer */
 	while (ll_func_rx_is_not_empty(spi)) {
 		(void) LL_SPI_ReceiveData8(spi);
 	}
 #endif
 
-	if (LL_SPI_GetMode(spi) == LL_SPI_MODE_MASTER) {
+	/* BSY flag is cleared when MODF flag is raised */
+	if (LL_SPI_IsActiveFlag_MODF(spi)) {
+		LL_SPI_ClearFlag_MODF(spi);
+	}
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	LL_SPI_ClearFlag_TXTF(spi);
+	LL_SPI_ClearFlag_OVR(spi);
+	LL_SPI_ClearFlag_SUSP(spi);
```

Comment:
The `LL_SPI_ClearFlag_SUSP` is cleared in `ll_func_disable_spi`.

Reply:
I wouldn't pay much attention to the `disable_spi` function as I believe is incorrect. It clears the `SUSP` flag because it suspends the transaction, which shouldn't be done unless there is a reason for it (and there isn't). So it shouldn't do either of the above, but that's out of this PR's scope.

Reply:
The less changes you do, the more this change is safe. Otrherwise make sure changes are done in dedicated commit with proper justification